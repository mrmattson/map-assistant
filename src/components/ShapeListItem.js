import React from 'react'
import PropTypes from 'prop-types'

import coordinateToString from '../helpers/coordinateToString'

function ShapeListItem ({ shape }) {
  // const coordinates = shape.geometry.coordinates[0]
  // let verticeButtonMsg = ''
  // if (this.state.showCoords) {
  //   verticeButtonMsg = 'Click to hide vertices'
  // } else {
  //   verticeButtonMsg = 'Click to show polygon vertices.'
  // }
  // <div><button onClick={this.showCoordinates}>{verticeButtonMsg}</button>
  //   {this.state.showCoords ? <pre>
  //     {coordinates.slice(0, coordinates.length - 1).map((coords, index) => {
  //       let str = ''
  //       if (index !== 0) {
  //         str += '\n\n'
  //       }
  //       str += `${index + 1}) ${coordinateToString(coords)}`
  //       return str
  //     })}
  //   </pre> : null}
  // </div>

  return (
    <li className="list-group-item">
      <div>{shape.properties.title}</div>
      <div>Center: <pre>{coordinateToString(shape.properties.centerCoordinates)}</pre></div>
      <div>Radius: {shape.properties.radius} {shape.properties.radiusUnits}</div>
    </li>
  )
}

// https://facebook.github.io/react/docs/typechecking-with-proptypes.html
ShapeListItem.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  shape: PropTypes.object.isRequired
}

export default ShapeListItem
