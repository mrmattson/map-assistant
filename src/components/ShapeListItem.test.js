/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ShapeListItem from './ShapeListItem'

import example from '../TestData/circles'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <ShapeListItem
      shape={example.circlePolygon}
    />, div)
})

it('clean snapshot', () => {
  const component = renderer.create(
    <ShapeListItem
      shape={{
        properties: {
          title: 'None',
          centerCoordinates: [0, 0],
          radius: 1,
          radiusUnits: 'units'
        }
      }}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('example circle polygon snapshot', () => {
  const component = renderer.create(
    <ShapeListItem
      shape={example.circlePolygon}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
