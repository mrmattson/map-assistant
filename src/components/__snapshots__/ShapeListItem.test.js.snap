exports[`test clean snapshot 1`] = `
<li
  className="list-group-item">
  <div>
    None
  </div>
  <div>
    Center: 
    <pre>
      00 00 00.0 N   000 00 00.0 E
    </pre>
  </div>
  <div>
    Radius: 
    1
     
    units
  </div>
</li>
`;

exports[`test example circle polygon snapshot 1`] = `
<li
  className="list-group-item">
  <div>
    Circle
  </div>
  <div>
    Center: 
    <pre>
      39 05 58.9 N   094 34 43.0 W
    </pre>
  </div>
  <div>
    Radius: 
    10
     
    nauticalMiles
  </div>
</li>
`;
